src/mawpy/utilities/validations.py

```diff
@@ -0,0 +1,15 @@
+import argparse
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def check_non_negative(value):
+    try:
+        f_value = float(value)
+    except Exception as ex:
+        logger.error(ex)
+        raise argparse.ArgumentTypeError(f"{value} is not a valid non-negative numerical value")
+    if f_value < 0:
+        raise argparse.ArgumentTypeError(f"{value} is not a valid non-negative numerical value")
+    return f_value
```

Comment:
There are 2 type of check that need to happen here and probably 2 different exception. Using this `argparse` functionality to check for negative value is not preferred and I would consider this very hacky. You should defer the negative value check after the value is parsed by argparse and leave this for type checking only..